File: src/main/kotlin/org/kite9/diagram/visualization/compaction/slideable/LoggingOptimisationStep.java
package org.kite9.diagram.visualization.compaction.slideable;

import org.kite9.diagram.visualization.compaction.AbstractCompactionStep;
import org.kite9.diagram.visualization.compaction.Compaction;
import org.kite9.diagram.visualization.compaction.Compactor;
import org.kite9.diagram.visualization.compaction.Embedding;
import org.kite9.diagram.visualization.display.CompleteDisplayer;

public class LoggingOptimisationStep extends AbstractCompactionStep {

	public LoggingOptimisationStep(CompleteDisplayer cd) {
		super(cd);
	}

	@Override
	public String getPrefix() {
		return "LOPS";
	}

	@Override
	public boolean isLoggingEnabled() {
		return true;
	}

	@Override
	public void compact(Compaction c, Embedding r, Compactor rc) {
		log.send("Embedding: "+r);
		optimise(c, c.getHorizontalSegmentSlackOptimisation(), c.getVerticalSegmentSlackOptimisation());
	}

	public void optimise(Compaction c, SegmentSlackOptimisation horizontalSegments, SegmentSlackOptimisation verticalSegments) {
		log.send(log.go() ? null : "Minimisation Steps: \n  HorizontalSegments: "+horizontalSegments.getSize()+" \n  Vertical Segments: "+verticalSegments.getSize()+" T: "+ (horizontalSegments.getSize() + verticalSegments.getSize()));
		log.send(log.go() ? null : "Push Steps: \n  Horizontal Segments: "+horizontalSegments.getPushCount()+ " \n  Vertical Segments: "+verticalSegments.getPushCount()+ "T: "+(horizontalSegments.getPushCount() + verticalSegments.getPushCount()));
		log.send("Horizontal Segments:", horizontalSegments.getAllSlideables());
		log.send("Vertical Segments:", verticalSegments.getAllSlideables());
	}
}

File: src/main/kotlin/org/kite9/diagram/visualization/compaction/slideable/LoggingOptimisationStep.kt
package org.kite9.diagram.visualization.compaction.slideable

import org.kite9.diagram.visualization.compaction.AbstractCompactionStep
import org.kite9.diagram.visualization.compaction.Compaction
import org.kite9.diagram.visualization.compaction.Compactor
import org.kite9.diagram.visualization.compaction.Embedding
import org.kite9.diagram.visualization.display.CompleteDisplayer

class LoggingOptimisationStep(cd: CompleteDisplayer) : AbstractCompactionStep(cd) {

    override val prefix: String
        get() = "LOPS"
    override val isLoggingEnabled: Boolean
        get() = true

    override fun compact(c: Compaction, r: Embedding, rc: Compactor) {
        log.send("Embedding: $r")
        optimise(c, c.getHorizontalSegmentSlackOptimisation(), c.getVerticalSegmentSlackOptimisation())
    }

    fun optimise(
        c: Compaction?,
        horizontalSegments: SegmentSlackOptimisation,
        verticalSegments: SegmentSlackOptimisation
    ) {
        log.send(
            if (log.go()) null else """Minimisation Steps: 
  HorizontalSegments: ${horizontalSegments.getSize()} 
  Vertical Segments: ${verticalSegments.getSize()} T: ${horizontalSegments.getSize() + verticalSegments.getSize()}"""
        )
        log.send(
            if (log.go()) null else """Push Steps: 
  Horizontal Segments: ${horizontalSegments.pushCount} 
  Vertical Segments: ${verticalSegments.pushCount}T: ${horizontalSegments.pushCount + verticalSegments.pushCount}"""
        )
        log.send("Horizontal Segments:", horizontalSegments.getAllSlideables())
        log.send("Vertical Segments:", verticalSegments.getAllSlideables())
    }
}